compiler/next/include/chpl/queries/query-impl.h

```diff
@@ -381,24 +444,56 @@ Context::querySetterUpdateResult(
 #define QUERY_GET_SAVED() \
   (BEGIN_QUERY_CONTEXT->queryGetSaved(BEGIN_QUERY_FOUND))
 
+/**
+  Use QUERY_BEGIN at the start of the implementation of a particular query.
+  It checks to see if an earlier result can be used and in that event returns
+  it.
+
+  Pass the name of the enclosing function as func, context is the
+  class Contex, and then pass any arguments to the query.
+ */
 #define QUERY_BEGIN(func, context, ...) \
   QUERY_BEGIN_INNER(false, func, context, __VA_ARGS__); \
   if (QUERY_USE_SAVED()) { \
     return QUERY_GET_SAVED(); \
   }
 
+/**
+  QUERY_BEGIN_INPUT is like QUERY_BEGIN but should be used
+  for input queries.
+ */
 #define QUERY_BEGIN_INPUT(func, context, ...) \
   QUERY_BEGIN_INNER(true, func, context, __VA_ARGS__) \
   if (QUERY_USE_SAVED()) { \
     return QUERY_GET_SAVED(); \
   }
 
-#define QUERY_ERROR(error) \
-  BEGIN_QUERY_CONTEXT->queryNoteError(error)
+/**
+  Returns a pointer to the partial result if the query is already running
```

Comment:
I think I would add that the partial report is stored by assigning to QUERY_CURRENT_RESULT and that this partial result is for another invocation of func that (up somewhere on the call stack) that was invoked with the exact values as arguments.